datadog/resource_datadog_synthetics_test_.go

```diff
@@ -1127,6 +1187,101 @@ func buildSyntheticsAPITestStruct(d *schema.ResourceData) *datadogV1.SyntheticsA
 
 	config.SetConfigVariables(configVariables)
 
+	if attr, ok := d.GetOk("api_step"); ok && syntheticsTest.GetSubtype() == "multi" {
+		steps := []datadogV1.SyntheticsAPIStep{}
+
+		for _, s := range attr.([]interface{}) {
+			step := datadogV1.SyntheticsAPIStep{}
+			stepMap := s.(map[string]interface{})
+
+			step.SetName(stepMap["name"].(string))
+			step.SetSubtype(datadogV1.SyntheticsAPIStepSubtype(stepMap["subtype"].(string)))
+
+			extractedValues := buildExtractedValues(stepMap["extracted_value"].([]interface{}))
+			step.SetExtractedValues(extractedValues)
+
+			assertions := stepMap["assertion"].([]interface{})
+			step.SetAssertions(buildAssertions(assertions))
+
+			request := datadogV1.SyntheticsTestRequest{}
+			requests := stepMap["request_definition"].([]interface{})
+			requestMap := requests[0].(map[string]interface{})
+			request.SetMethod(datadogV1.HTTPMethod(requestMap["method"].(string)))
+			request.SetUrl(requestMap["url"].(string))
+			request.SetBody(requestMap["body"].(string))
+			request.SetTimeout(float64(requestMap["timeout"].(int)))
+
+			requestHeaders := stepMap["request_headers"].(map[string]interface{})
+			if len(requestHeaders) > 0 {
+				request.SetHeaders(make(map[string]string))
+			}
+			for k, v := range requestHeaders {
+				request.GetHeaders()[k] = v.(string)
+			}
+
```

Comment:
Let's build the map first, then set it.
Also we already know in advance the length of the map, so let's use that in the make.